snakebids/app.py

```diff
@@ -253,14 +254,20 @@ def update_config(config: dict[str, Any], snakebids_args: SnakebidsArgs) -> None
     for input_type in pybids_inputs.keys():
         arg_filter_dict = args[f"filter_{input_type}"]
         if arg_filter_dict is not None:
-            pybids_inputs[input_type]["filters"].update(arg_filter_dict)
+            pybids_inputs[input_type].setdefault("filters", {})
+            for entity, filter_ in arg_filter_dict.items():
+                if filter_ == OptionalFilter:
```

Comment:
I think we want `is` here
```suggestion
                if filter_ is OptionalFilter:
```